mlflow/server/js/src/experiment-tracking/components/MetricsPlotPanel.js

```diff
@@ -436,6 +439,40 @@ export class MetricsPlotPanel extends React.Component {
     };
     this.updateUrlState({ layout: mergedLayout, lastLinearYAxisRange });
   };
+  
+  convertJsonToCsv = () => {
+    const metrics = this.getMetrics();
+    //const parsedJson = JSON.parse(metrics);
+    //console.log("Parsed JSON: " + parsedJson)
+    const selectedMetricKeys = this.getUrlState().selectedMetricKeys;
+    console.log("selected metrics: " + selectedMetricKeys);
+    const heading = Object.keys(metrics[0].history[0]).join(",") + "\n"
+    const metricValues = selectedMetricKeys.map(
+      (metricKey) => {
+        return metrics
+          .filter((metric) => metric.metricKey === metricKey)
+          .map((metric) => {
+            return metric.history
+              .map((metricValue) => {
+                return Object.values(metricValue);
+              }).join("\n");
+          });
+      }).join("\n");
+
+    /*const body = metrics.metricKey
+      .map((metric) => {
+        metric.map((j) => Object.values(j).join(",")).join("\n");
+    })*/
+    return `${heading}${metricValues}`;
+  };
+
+  onDownloadCsv = () => {
+    const csv = this.convertJsonToCsv();
+    console.log("CSV: " + csv)
+    const blob = new Blob([csv], { type: 'application/csv;charset=utf-8' });
+    console.log("Blob: " + blob)
+    saveAs(blob, 'metrics.csv');
```

Comment:
should the filepath have a unique identifier so that users aren't inadvertently overwriting or creating named index copies of the same base name? `metrics.csv` seems confusing as well since it's commonly what a run_id extract of metrics would be called.

Reply:
nice catch. from my tests, the browser automatically adds (x) to the end of the file to prevent that. could also add a unique idenfifier - what would be the best in this case - something like metrics - HH-mm-ss_dd-mm-yyyy? or maybe a GUID?

Reply:
all right, so I've added the run_id to the filename, e.g. metrics_run_id_XXXXXXXXXXX.csv